src/Compiler/DependencyManager/DependencyProvider.fs

```diff
@@ -152,14 +152,25 @@ type ReflectionDependencyManagerProvider
         resolveDepsExWithScriptInfoAndTimeout: MethodInfo option,
         clearResultCache: MethodInfo option,
         outputDir: string option,
-        useResultsCache: bool
+        useResultsCache: bool,
+        sdkDirOverride: string option
     ) =
 
     let instance =
-        if not (isNull (theType.GetConstructor([| typeof<string option>; typeof<bool> |]))) then
-            Activator.CreateInstance(theType, [| outputDir :> objnull; useResultsCache :> objnull |])
-        else
-            Activator.CreateInstance(theType, [| outputDir :> objnull |])
+        let arguments: (Type * objnull) array =
+            [|
+                typeof<string option>, outputDir
+                typeof<bool>, useResultsCache
+                typeof<string option>, sdkDirOverride
+            |]
+
+        let n = arguments.Length
+
+        seq { for i in n - 1 .. -1 .. 0 -> arguments[0..i] }
```

Comment:
I get the intention.

But still, I believe future human and non-human coders would appreciate a few comments about what is being done here and especially why.